src/api/routes/offer.js

```diff
@@ -28,6 +29,25 @@ module.exports = (app) => {
         }
     });
 
+    /**
+     * Gets an offer from the database with its id
+    */
+    router.get("/:offer", validators.get, async (req, res, next) => {
+        try {
+            const offer = await (new OfferService()).getOfferById(req.params.offer);
+            if (offer !== null) {
+                return res.json(offer);
+            } else {
+                return res.status(HTTPStatus.UNAUTHORIZED).json({
+                    reason: `No offer with id:${req.params.offer}`,
+                });
+            }
```

Comment:
It's usually a good practice to improve readability to have only the error case in the if block, and the happy path outside, like so:

```js
            if (offer === null) {
                    res.status(HTTPStatus.UNAUTHORIZED).json({
                    reason: `No offer with id:${req.params.offer}`,
                });
            }
            return res.json(offer);
```

Reply:
Additionally, shouldn't this be a simple `!offer` instead of a strict check with null? @imnotteixeira

Reply:
Mongoose API states that when there is no match it will return null, however we can play it safe and use a !offer check instead